### ISSUE TITLE
Az Backup Policy Set - Processing with Az Cli

### ISSUE THREAD

**Issue by steharp** [Azure/azure-cli#14280]
## Describe the bug

**Command Name**
`az backup policy set`

**Errors:**
```
The variable passed should be in valid JSON format and be supplied by az backup CLI commands.
            Make sure that you use output of relevant 'az backup show' commands and the --out is 'json'
            (use -o json for explicit JSON output) while assigning value to this variable.
            Take care to edit only the values and not the keys within the JSON file or string.
            
Traceback (most recent call last):
pip-install-yf22vmg2\knack\knack\cli.py, ln 215, in invoke
cli\core\commands\__init__.py, ln 654, in execute
...
            Make sure that you use output of relevant 'az backup show' commands and the --out is 'json'
            (use -o json for explicit JSON output) while assigning value to this variable.
            Take care to edit only the values and not the keys within the JSON file or string.
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az backup policy set -g {} --subscription {} -v {} --policy {}`

$RG = 'myry'
$subid = az account show --query id --output tsv

az backup policy set -g $RG --subscription $subid -v weeklybackupvault --policy '{"eTag":null,,"location":null,"name":"TestWeeklySchedule","properties":{"backupManagementType":"AzureIaasVM","instantRpRetentionRangeInDays","protectedItemsCount":0,"retentionPolicy":{"dailySchedule":{"retentionDuration":{"count":8,"durationType":"Weeks"},"retentionTimes":["2019-08-18T20:00:00Z"]},"monthlySchedule":null,"retentionPolicyType":"LongTermRetentionPolicy","weeklySchedule":null,"yearlySchedule":null},"schedulePolicy":{"schedulePolicyType":"SimpleSchedulePolicy","scheduleRunDays":["Friday"],"scheduleRunFrequency":"Weekly","scheduleRunTimes":["2019-08-18T20:00:00Z"],"scheduleWeeklyFrequency":1},"timeZone":"UTC"},"resourceGroup":"$RG","tags":null,"type":"Microsoft.RecoveryServices/vaults/backupPolicies"}'

## Expected Behavior

deploy a backup policy for IaaS VM.

Seem like there is a complete lack of detailed Az Cli detailed examples for this.

## Environment Summary
```
Windows-10-10.0.18362-SP0
Python 3.6.6
Installer: MSI

azure-cli 2.6.0 *

Extensions:
aks-preview 0.4.45
azure-devops 0.18.0
azure-firewall 0.5.0

```

**Comment by yonzhan** [comment]
backup

**Comment by yungezz** [comment]
hi @steharp  the error is due to the input policy string isn't a valid json string. Could you pls  fix the format and try again? thanks
```
{"eTag":null,,"location":null,"name":"TestWeeklySchedule","properties":{"backupManagementType":"AzureIaasVM","instantRpRetentionRangeInDays","protectedItemsCount":0,"retentionPolicy":{"dailySchedule":{"retentionDuration":{"count":8,"durationType":"Weeks"},"retentionTimes":["2019-08-18T20:00:00Z"]},"monthlySchedule":null,"retentionPolicyType":"LongTermRetentionPolicy","weeklySchedule":null,"yearlySchedule":null},"schedulePolicy":{"schedulePolicyType":"SimpleSchedulePolicy","scheduleRunDays":["Friday"],"scheduleRunFrequency":"Weekly","scheduleRunTimes":["2019-08-18T20:00:00Z"],"scheduleWeeklyFrequency":1},"timeZone":"UTC"},"resourceGroup":"$RG","tags":null,"type":"Microsoft.RecoveryServices/vaults/backupPolicies"}
```

**Comment by steharp** [comment]
Hi thanks for your comment:  Im aware its a syntax error, however I cannot gather the correct syntax, find any use-full documentation, even when following this will templates, they lack the required information to build a successful policy.

It is most frustrating that guidance from Microsoft is non-existent in this area, anyone who has successfully scripted with Az Cli or via ARMTemplate who could share a working example would be great.

**Comment by yungezz** [comment]
@steharp thanks for sharing the detail. it's valid painpoint. we're routing the issue to appropriate team to investigate.

**Comment by pvrk** [comment]
@yungezz : Sorry for this experience. We will fix the documentation with the right examples soon.

**Comment by pvrk** [comment]
@steharp : Just for information. To get the policy JSON, you can use the [az backup default policy for VM](https://docs.microsoft.com/cli/azure/backup/policy?view=azure-cli-latest#az-backup-policy-get-default-for-vm) command, route it to a file, then edit with necessary values. In this way, you need worry about syntax.

**Comment by mepand** [comment]
[Sample policy.docx](https://github.com/Azure/azure-cli/files/5073830/Sample.policy.docx)

@steharp Please refer the sample VM policy in the attached document. Copy it in a file and save it as Policy.json. Then execute the below command .

az backup policy set --resource-group rg1234 --vault-name testvault --policy C:\temp2\Policy.json --name testing123

[You can modify the policy path in the above command accordingly]

I am working on updating the documentation as well with a similar example.

**Comment by mepand** [comment]
@steharp Could you please confirm if the steps I mentioned above unblocked you?

**Comment by mepand** [comment]
Documentation update has been checked in. Changes would be live by tomorrow. Since there is no further action pending on the PG side, closing the issue.

**Comment by wi5nia** [comment]
And what about having this policy file read into a variable for example in PowerShell? Is it even possible because I can't get it to work @pvrk @mepand ?

I have the policy in a .json file and when I pass the path to that file in --policy all is good. But for example if I do
$variable = Get-Content policy.json

and then pass it to the --policy as
--policy $variable
or
--policy '$variable'
or
--policy "$variable"

In all cases gives an error that it's not a valid json :/

**Comment by wi5nia** [comment]
I have another issue in this realm. I have a policy which I have in a file. Let's call it **policy.json**. 
When I provide it to the 
`az backup policy create --policy policy.json ...`
all works great.

But now in PowerShell Desktop I am doing something like:
```
$policy = Get-Content policy.json
$policy = $policy.Replace("UTC", "W. Europe Standard Time")
az backup policy create "$policy" ...
```

I get an error"
` az : ERROR: unrecognized arguments: Standard Time   }, ...`

Any ideas, suggestions @mepand @pvrk ?
